First section of the tutorial 1/3 (#505)
open-telemetry/opentelemetry-java-instrumentation: instrumentation-core/spring/README.md

```java
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpMethod;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

import io.opentelemetry.context.propagation.HttpTextFormat;
import io.opentelemetry.trace.Span;
import io.opentelemetry.trace.SpanContext;
import io.opentelemetry.trace.Tracer;

@Component
public class HttpUtils {

  @Autowired
  private Tracer tracer;

  private HttpTextFormat<SpanContext> textFormat;
  private HttpTextFormat.Setter<HttpHeaders> setter;

  public HttpUtils(Tracer tracer) {
    textFormat = tracer.getHttpTextFormat();
    setter = new HttpTextFormat.Setter<HttpHeaders>() {
      @Override
      public void put(HttpHeaders headers, String key, String value) {
        headers.set(key, value);
      }
    };
  }

  public String callEndpoint(String url) throws Exception {
    HttpHeaders headers = new HttpHeaders();

>     Span currentSpan = tracer.getCurrentSpan();
>     textFormat.inject(currentSpan.getContext(), headers, setter);

Reviewer: I think this API has changed (probably worth re-checking all of the code against `0.5.0`)
```suggestion
    textFormat.inject(Context.current(), headers, setter);
```

import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class SecondServiceApplication {

  public static void main(String[] args) throws IOException {
    SpringApplication.run(SecondServiceApplication.class, args);
  }
}
```

4. Create a RestController for SecondService
5. Start a span to wrap the SecondServiceController

```java
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import io.opentelemetry.context.Scope;
import io.opentelemetry.trace.Span;
import io.opentelemetry.trace.Tracer;

@RestController
@RequestMapping(value = "/time")
public class SecondServiceController {
  @Autowired
  private Tracer tracer;

  @GetMapping
  public String callSecondTracedMethod() {
    Span span = tracer.spanBuilder("time").startSpan();
    span.addEvent("SecondServiceController Entered");
    span.setAttribute("what.am.i", "Tu es une legume");

    try (Scope scope = tracer.withSpan(span)) {
      return "It's time to get a watch";

>     } finally {

Reviewer: it's not clear the reason for having a scope here

      span.end();
    }
  }
}
```

### Run FirstService and SecondService

***To view your distributed traces ensure either LogExporter or Jaeger is configured in the OtelConfig.java file*** 

To view traces on the Jaeger UI, deploy a Jaeger Exporter on localhost by running the command in terminal:

`docker run --rm -it --network=host jaegertracing/all-in-one` 

After running Jaeger locally, navigate to the url below. Make sure to refresh the UI to view the exported traces from the two web services:

`http://localhost:16686`
 

> Run FirstService and SecondService from command line or using an IDE. The end point of interest for FirstService is `http://localhost:8080/message` and  `http://localhost:8081/time` for SecondService. Entering `localhost:8080/time` in a browser should call FirstService and then SecondService, creating a trace. To send a sample request enter the following in a browser of your choice:

Reviewer: i'm not sure, i think the url is not correct, and the suggestion to remove the word `sample` is just to avoid confusion with sampling
```suggestion
Run FirstService and SecondService from command line or using an IDE. The end point of interest for FirstService is `http://localhost:8080/message` and  `http://localhost:8081/time` for SecondService. Entering `localhost:8080/message` in a browser should call FirstService and then SecondService, creating a trace. To send a request enter the following in a browser of your choice:
```

> `http://localhost:8080/time`

Reviewer: i'm not sure, i think the url is not correct
```suggestion
`http://localhost:8080/message`
```

> ***Note: The default port for the Apache Tomcat is 8080. On localhost both FirstService and SecondService services will attempt to run on this port raising an error. To avoid this add `server.port=8081` to the resources/application.properties file. Ensure the port specified corresponds to port referenced by FirstServiceController.SS_URL. ***

Reviewer: ```suggestion
***Note: The default port for the Apache Tomcat is 8080. On localhost both FirstService and SecondService services will attempt to run on this port raising an error. To avoid this add `server.port=8081` to the resources/application.properties file. Ensure the port specified corresponds to port referenced by FirstServiceController.SECOND_SERVICE_URL. ***
```

import org.springframework.web.bind.annotation.RestController;

import io.opentelemetry.context.Scope;
import io.opentelemetry.trace.Span;
import io.opentelemetry.trace.Tracer;

@RestController
@RequestMapping(value = "/time")
public class SecondServiceController {
   @Autowired
   private Tracer tracer;

   @GetMapping
   public String callSecondTracedMethod() {
      Span span = tracer.spanBuilder("time").startSpan();
      span.addEvent("SecondServiceController Entered");
      span.setAttribute("what.am.i", "Tu es une legume");

      try{
         return "It's time to get a watch";
      } finally {
         span.end();
      }
   }
}
```

### Run FirstService and SecondService

***To view your distributed traces ensure either LogExporter or Jaeger is configured in the OtelConfig.java file*** 

To view traces on the Jaeger UI, deploy a Jaeger Exporter on localhost by running the command in terminal:

`docker run --rm -it --network=host jaegertracing/all-in-one` 

After running Jaeger locally, navigate to the url below. Make sure to refresh the UI to view the exported traces from the two web services:

`http://localhost:16686`

> Run FirstService and SecondService from command line or using an IDE. The end point of interest for FirstService is `http://localhost:8080/message` and  `http://localhost:8081/time` for SecondService. Entering `localhost:8080/message` in a browser should call FirstService and then SecondService, creating a trace. To send a sample request enter the following in a browser of your choice:
> 
> `http://localhost:8081/time`

Reviewer: It's not clear to me what the part I removed below is for, so I suggest either removing or clarifying that it

```suggestion
Run FirstService and SecondService from command line or using an IDE. The end point of interest for FirstService is `http://localhost:8080/message` and  `http://localhost:8081/time` for SecondService. Entering `localhost:8080/message` in a browser should call FirstService and then SecondService, creating a trace.
```
